Maven compatibility

keinhaar opened the pull request:
The nobuto build is just to rebuild the maven.compatibility.jar itself. It is not needed for the plugin build, because the precompiled jar is already in the apiclient plugin.

protoism commented:
Hi @keinhaar,
This PR can't make it in as it is now.

1) No reference to Nobuto should enter into gwt-eclipse-plugin
2) Conflicts must be resolved

Moreover, can you clarify the origin of this "maven-compatibility.jar" artifact?

keinhaar commented:
@protoism 
I've rebased and removed nobuto.
The maven.compatibility.jar is made by myself. It's just a wrapper that makes the correct maven calls depending on the used eclipse version. 
The sources are still included in the commit, but can no longer be build, because it needs **old and a new eclipse dependencies** to compile. I don't know how to do that with maven.

protoism commented on maven.compatibility/source/java/maven/compatibility/ProjectConfigurationRequestWrapper.java:
```diff
@@ -0,0 +1,51 @@
+
+public class ProjectConfigurationRequestWrapper
+{
+    public static IProject getProject(ProjectConfigurationRequest request)
```
I wonder if this is the way to go... 
Using reflection is not that complicate.
See here:

https://github.com/groovy/groovy-eclipse/blob/master/extras/org.codehaus.groovy.m2eclipse/src/org/codehaus/groovy/m2eclipse/GroovyProjectConfigurator.java

niloc132 replied:
I have to agree at least in part - I don't object to a new project helping with this, but it should be buildable rather than checking in the jar directly.

protoism commented on plugins/com.gwtplugins.gdt.eclipse.apiclientlib/.classpath:
```diff
@@ -1,38 +1,39 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
This might be an EOL change, or a spacing change, but they must be avoided, or done in an ad-hoc commit, as they make reviewing way more complicate

niloc132 replied:
Click the "gear" icon in the diff view, and then select "hide whitespace", or just add ?w=1 to the url

protoism reviewed (changes requested):
I don't agree with this PR. The job could be done easily with reflection, and we would not introduce a binary dependency on an artifact which is "not easy to build".

protoism commented:
I tried to achieve the same results with reflection here: https://github.com/gwt-plugins/gwt-eclipse-plugin/pull/451
